### PR TITLE
New: Add modules feature flag (refs #72)

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

**Comment by nzakas** [comment]
This basically switches to using `ecmaFeatures.modules` for enabling module features.

Still a bunch of tests missing (for classes, strict mode, etc.), but this should be a good foundation to build upon.

File: tests/lib/modules.js [hunk 1]
```diff
@@ -1,148 +0,0 @@
-/**
```
Comment (xjamundx) [inline tests/lib/modules.js]: Love putting the modules back in ecma-features!

Comment (nzakas) [inline tests/lib/modules.js]: Really that was @caridy's original idea. Just took me a while to see that he was right. :)

**Comment by xjamundx** [comment]
Do you want to sneak something like https://github.com/caridy/espree/pull/1in here as well?

**Comment by nzakas** [comment]
Was hoping I could get you to do that after I merge this :)

**Comment by xjamundx** [comment]
Sure!

**Comment by nzakas** [comment]
Okay, all ready for you!

**Pull request merged**